# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/TextureRecorded.cpp
# Commit: c8142bb80fe0
# Full Hash: c8142bb80fe0e75dff9587c999e6faf25d8760b9
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2023-09-16 09:14:45
# Description:
#   Bug 1853178: Return null if we have no DrawTarget in RecordedTextureData::BorrowSnapshot. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D188347
# ==============================================================================

diff -r d374e55d4625 -r c8142bb80fe0 gfx/layers/client/TextureRecorded.cpp
--- a/gfx/layers/client/TextureRecorded.cpp	Fri Sep 15 16:54:45 2023 +0000
+++ b/gfx/layers/client/TextureRecorded.cpp	Fri Sep 15 17:17:02 2023 +0000
@@ -95,7 +95,11 @@
 }
 
 already_AddRefed<gfx::SourceSurface> RecordedTextureData::BorrowSnapshot() {
-  MOZ_ASSERT(mDT);
+  // There are some failure scenarios where we have no DrawTarget and
+  // BorrowSnapshot is called in an attempt to copy to a new texture.
+  if (!mDT) {
+    return nullptr;
+  }
 
   if (mSnapshot) {
     return mCanvasChild->WrapSurface(mSnapshot);
